### DISCUSSION TITLE
Fehlerhafte Darstellung in Chrome und Safari, ggfs. in weiteren Browsern

### DISCUSSION THREAD

**Discussion by chfink0** [OpenCCU/OpenCCU#2588] [🇩🇪 Fragen&Antworten] [answered]
### Describe the issue you are experiencing

Viele Menüs bleiben in o.g. Desktopbrowsern leer oder werden nicht ausreichend gefüllt. Zwei Beispiele im Video unten: 
Es lassen sich keine Geräteeigenschaften anzeigen
Neue Geräte können nicht hinzugefügt werden, man sieht sie im Posteingang, aber nicht nach dem Klicken in die Posteingangsliste.
Ein Neustart bietet keine Abhilfe, bei Ersteinrichtung (danach eventuell Updates?) lief alles. Auf dem Iphone wird es korrekt angezeigt. Jungfräulicher Chrome mit daher leerem Cache konnte keine Abhilfe schaffen.

https://github.com/jens-maus/RaspberryMatic/assets/120484617/4ee65662-fa07-44b2-a191-14e3eb65abe4

### Describe the behavior you expected

Funktionsfähiges Menü

### Steps to reproduce the issue

1. Raspberrymatic bzw. Home Assistant neustarten
2. Geräte aufrufen
3. Auf ein Gerät klicken
...

### What is the version this bug report is based on?

3.73.9.20231130

### Which base platform are you running?

rpi4 (RaspberryPi4)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
Gebe gerne weitere Infos, müsste nur wissen, wonach ich schauen muss. Danke!
```

**Comment by Baxxy13** [comment]
Browserkonsole mit (i.d.R.) F12 öffnen und nach Meldungen Ausschau halten bzw. diese zeigen. 

Ich nutze nur "Chromium-based-Browser" und Firefox, da kann ich aktuell keine Probleme erkennen.

**Reply by chfink0** [reply]
Danke für den Hinweis, Meldungen in der Browserkonsole, die ich zuordnen könnte, finde ich nicht.

Allerdings ist mir dank dieses Testens aufgefallen, dass das Problem lediglich dann besteht, wenn ich über "die Ferne" a.k.a. Cloudflare zugreife. Alle anderen Funktionen, sowohl in Raspberrymatic als auch in Home Assistant sind allerdings absolut fehlerfrei... 

Wäre durchaus sehr schade, wenn man Temperaturprofile - also auch Feriensteuerung an und aus - nicht aus der Ferne machen könnte.

**Reply by jens-maus** [reply]
> Allerdings ist mir dank dieses Testens aufgefallen, dass das Problem lediglich dann besteht, wenn ich über "die Ferne" a.k.a. Cloudflare zugreife. Alle anderen Funktionen, sowohl in Raspberrymatic als auch in Home Assistant sind allerdings absolut fehlerfrei...
> Wäre durchaus sehr schade, wenn man Temperaturprofile - also auch Feriensteuerung an und aus - nicht aus der Ferne machen könnte.

Beschreib mal dein konkretes Nutzungsszenario via "Cloudflare". Wie ist das umgesetzt, etc. Das optimale Mittel der Wahl um "von unterwegs" bzw. "der Ferne" auf seine CCU WebUI zuzugreifen ist und sollte eine VPN Verbindung sein und nicht irgendwelche Portweiterleitungen oder ähnliches....

**Reply by Baxxy13** [reply]
Vermutlich Cloudflare-Tunnel, Zugriff über eigene Domain.
Ist quasi auch VPN ohne Portfreigaben. Nutze ich neuerdings auch um auf den HA von unterwegs zuzugreifen.
Wollte das erst über den Reverse-Proxy von CloudMatic machen, aber nach den letzten Erkenntnissen bezüglich Sicherheit bei CM hab ich mich dagegen entschieden.  

Da stellt sich mir die Frage was das für eine RaspberryMatic ist auf dem Pi4B. 
Spekulation: HA-AddOn.

**Reply by jens-maus** [reply]
So wird es vmtl. sein. Dann sollte man da eben nicht über die Ingress-Seite zugreifen, sondern direkt über einen alternativen Port (z.B. 8080). Oder es sagt mir einer was das Problem ist bzw. zeigt mir die Fehler in der Browserconsole und dann kann ich den HA-Proxy ggf. anpassen das er auch die URLs von Cloudflare entsprechend berücksichtigt...

**Reply by chfink0** [reply, marked as answer]
So ist es, wie Ihr vermutet, es ist das AddOn, ist allerdings ein i386 mit home assistant drauf. Die VPN Verbindung würde es auch tun - bei mir leider im konkreten Fall problematisch, da mein ISP nur ipv6 anbietet und mit wireguard ist es leider ein kampf, zumindest in Kombination mit Mobilfunk.. Das ist ansonsten aber ne feine und recht sichere Sache mit dem Cloudflare und eigener Domain, (alles kostenfrei) auch die Familie schafft damit den Zugang zu Home Assistant ;-) Ich verstehe aber, dass bei mir wohl ein sehr seltener Verbindungsfall vorliegt, der sicherlich nicht viele betrifft. Vermutlich gebe ich mich noch mal einem Versuch mit VPN hin, das ist mit Hinblick auf die Zukunft und Themen fernab von Home Assistant und Raspberrymatic ne sinnvolle Sache. Insofern vielen Dank für Euren Support, Jens Maus und Baxxy13, die Lösung für mich sollte dann eher VPN sein, alles andere lohnt vermutlich den Aufwand nicht.

**Reply by Baxxy13** [reply]
Die Frage wäre noch warum du "aus der Ferne" auf die WebUI musst.
Bei mir ist alles wichtige, was ggf. ab und zu händisch bedient werden muss, im HA-Dashboard drin. 
Das lässt sich mittels der HA-App über den Cloudflare-Tunnel problemlos bedienen.

Wusste gar nicht das Wireguard bei IPv6 strauchelt, was ist mit Tailscale als Alternative?

**Reply by jens-maus** [reply]
@Baxxy13 Kannst du die Probleme bzgl. Cloudflare und HA-Addon WebUI nachvollziehen? Vielleicht ist der Fix im HA-Proxy doch einfacher, dann könnte ich mir das anschauen...

**Reply by Baxxy13** [reply]
Aktuell kann ich da nichts prüfen da ich zwar den Nightly als AddOn ggf. starten könnte, der aber weder Funkmodul noch Geräte hat. 
Muss ich mal gucken ob ich hier was für einen Test umswitchen kann.

**Reply by jens-maus** [reply]
würde vmtl. reichen wenn du einfach da ne config restores mit geräten um das zu simulieren. da braucht es vmtl. gar kein echtes funkmodul...

**Reply by Baxxy13** [reply]
Hab das nun schnell mal mit nem HmIP-RFUSB am AddOn getestet.
Über Cloudflare-Tunnel auf den HA / RM zugegriffen und eine HmIP-PSM angelernt und anschließend bedient.
Läuft hier alles wie es soll.

**Reply by chfink0** [reply]
Vielen Dank fürs Testen, eigenartig dass es dann nur bei mir nicht funktioniert... Werde mal noch weiter testen, wenn sich was gewinnbringendes ergibt, gebe ich Bescheid. Grund warum ich aus der Ferne auf die WebUI müsste wäre lediglich um nen Urlaubsmodus bzw. alle Wochenprogramme aus- oder auf ein anderes WP zu schalten. Ansonsten wird sich wohl, wenn mal eingerichtet und angelegt ist, dort den Rest des Jahres nicht sonderlich viel tun.
